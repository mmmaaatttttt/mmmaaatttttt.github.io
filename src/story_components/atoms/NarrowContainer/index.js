import styled from "styled-components";
import PropTypes from "prop-types";
import media, { sizes } from "utils/media";

const NarrowContainer = styled.div`
  width: ${props => props.width};
  margin: 0 auto;

  ${props => media[props.fullWidthAt]`
    width: 100%;
  `};
`;

NarrowContainer.propTypes = {
  width: PropTypes.string.isRequired,
  fullWidthAt: PropTypes.oneOf(Object.keys(sizes))
};

NarrowContainer.defaultProps = {
  width: "80%",
  fullWidthAt: "extraSmall"
};

export default NarrowContainer;
