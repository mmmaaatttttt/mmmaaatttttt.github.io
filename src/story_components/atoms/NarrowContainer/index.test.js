import React from "react";
import { render } from "@testing-library/react";
import NarrowContainer from "./";

it("renders without crashing", () => {
  render(<NarrowContainer />);
});

it("matches snapshot with props and children", () => {
  const { asFragment } = render(
    <NarrowContainer width="75%" fullWidthAt="small">
      <div>hello</div>
      <div>goodbye</div>
    </NarrowContainer>
  );
  expect(asFragment()).toMatchSnapshot();
});
