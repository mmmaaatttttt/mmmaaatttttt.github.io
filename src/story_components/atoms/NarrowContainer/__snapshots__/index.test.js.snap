// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children when passed in 1`] = `
<StyledComponent
  $$typeof={Symbol(react.element)}
  _owner={null}
  _store={Object {}}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  width: ",
          [Function],
          ";
  margin: 0 auto;

  ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "fullWidthAt": "extraSmall",
        "width": "80%",
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "fullWidthAt": [Function],
        "width": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  fullWidthAt="extraSmall"
  props={
    Object {
      "children": Array [
        <div>
          hello
        </div>,
        <div>
          goodbye
        </div>,
      ],
      "fullWidthAt": "small",
      "width": "75%",
    }
  }
  type={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  width: ",
          [Function],
          ";
  margin: 0 auto;

  ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "fullWidthAt": "extraSmall",
        "width": "80%",
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "fullWidthAt": [Function],
        "width": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  width="80%"
/>
`;

exports[`renders props when passed in 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  width: ",
          [Function],
          ";
  margin: 0 auto;

  ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "fullWidthAt": "extraSmall",
        "width": "80%",
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "fullWidthAt": [Function],
        "width": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  fullWidthAt="small"
  width="50%"
/>
`;

exports[`renders successfully 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  width: ",
          [Function],
          ";
  margin: 0 auto;

  ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "fullWidthAt": "extraSmall",
        "width": "80%",
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "fullWidthAt": [Function],
        "width": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  fullWidthAt="extraSmall"
  width="80%"
/>
`;
